parlai/core/torch_generator_agent.py

```diff
@@ -617,14 +642,19 @@ def beam_search(self, model, batch, beam_size, start=1, end=2,
 
             score, incr_state = model.decoder(decoder_input, encoder_states, incr_state)
             # only need the final hidden state to make the word prediction
-            score = score[:, -1, :]
+            score = score[:, -1, :].unsqueeze(1)
```

Comment:
This also seems surprising to me, that the bug is here. I'll need to investigate further.

Reply:
steps to get this thing happened:

1. train the model like this (should be newlines there...):
training:
`python train.py \
    -dt train \
    -t convai2:self \
    -m seq2seq \
    -bs 128 \
    -dp "${DATA_PATH}" \
    --model-file "${RUN_DIR}"/model \
    --dict-file "${DATA_PATH}"/convai2.dict \
    --numlayers 2 \
    --hiddensize 512 \
    --dropout 0.5 \
    --optimizer adam \
    -lr 0.001 \
    -esz 300 \
    --attention general \
    -lt enc_dec \
    -emb glove \
    -veps 5`

2. test the model:
`python eval_model.py \
    -dt valid \
    -t convai2:self \
    -m seq2seq \
    -bs 128 \
    -dp "${DATA_PATH}" \
    --model-file PATH \
    --dict-file "${DATA_PATH}"/convai2.dict \
    --beam-size 2 \`